src/scripts/main.js

```diff
@@ -1,3 +1,227 @@
 'use strict';
 
-// write code here
+function rowToObj(row) {
+  const cells = row.querySelectorAll('td');
+
+  return {
+    name: cells[0]?.textContent.trim() || null,
+    position: cells[1]?.textContent.trim() || null,
+    office: cells[2]?.textContent.trim() || null,
+    age: Number(cells[3]?.textContent.trim()) || null,
+    salary: parseFloat(cells[4]?.textContent.replace(/[$,]/g, '')) || null,
+  };
+}
+
+function sortTableArr(a, b, col, direction) {
+  if (a[col] < b[col]) {
+    return direction === 'asc' ? -1 : 1;
+  }
+
+  if (a[col] > b[col]) {
+    return direction === 'asc' ? 1 : -1;
+  }
+
+  return 0;
+}
+
+function sortTable(element) {
+  if (element.tagName === 'TH') {
+    const col = element.textContent.toLowerCase();
```

Comment:
Using `element.textContent.toLowerCase()` as the column key for sorting might not work if the column names do not match the object keys in `rowToObj`. Consider mapping column names to object keys explicitly.